File: app/src/main/java/ru/geekbrains/notes/data/CardSourceLocalImplementation.java
package ru.geekbrains.notes.data;

import android.content.res.Resources;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

import ru.geekbrains.notes.R;

public class CardSourceLocalImplementation implements CardSource {
    private List<CardData> dataSource;
    private final Resources resources;

    @Override
    public int size() {
        return this.dataSource.size();
    }

    @Override
    public CardData getCardData(int position) {
        return this.dataSource.get(position);
    }

    @Override
    public void deleteCardData(int position) {
        dataSource.remove(position);
    }

    @Override
    public void updateCardData(int position, CardData newCardData) {
        dataSource.set(position, newCardData);
    }

    @Override
    public void addCardData(CardData newCardData) {
        dataSource.add(newCardData);
    }

    @Override
    public void clearCardData() {
        dataSource.clear();
    }

    public CardSourceLocalImplementation(Resources resources) {
        this.resources = resources;
    }

    @Override
    public CardSource init(CardsSourceResponse cardsSourceResponse) {
        dataSource = new ArrayList<>();
        String[] titles = resources.getStringArray(R.array.notes);
        String[] texts = resources.getStringArray(R.array.texts);

        for (int i = 0; i < titles.length; i++) {
            dataSource.add(new CardData(titles[i], texts[i], Calendar.getInstance().getTime()));
        }

        if(cardsSourceResponse != null) {
            cardsSourceResponse.initialized(this);
        }

        return this;
    }
}

File: app/src/main/java/ru/geekbrains/notes/repository/CardSourceLocalImplementation.kt
package ru.geekbrains.notes.repository

import android.content.res.Resources
import ru.geekbrains.notes.R
import ru.geekbrains.notes.model.CardData
import java.util.*

class CardSourceLocalImplementation(private val resources: Resources) :
    CardSource {
    private var dataSource: MutableList<CardData?>? = null
    override fun size(): Int {
        return dataSource!!.size
    }

    override fun getCardData(position: Int): CardData {
        return dataSource!![position]!!
    }

    override fun deleteCardData(position: Int) {
        dataSource!!.removeAt(position)
    }

    override fun updateCardData(position: Int, newCardData: CardData) {
        dataSource!![position] = newCardData
    }

    override fun addCardData(newCardData: CardData) {
        dataSource!!.add(newCardData)
    }

    override fun clearCardData() {
        dataSource!!.clear()
    }

    override fun getCards(cardsSourceResponse: CardsSourceResponse): CardSource {
        dataSource = ArrayList()
        val titles = resources.getStringArray(R.array.notes)
        val texts = resources.getStringArray(R.array.texts)
        for (i in titles.indices) {
            (dataSource as ArrayList<CardData?>).add(
                CardData(
                i.toString(), titles[i], texts[i], Calendar.getInstance().time)
            )
        }
        cardsSourceResponse.initialized(this)
        return this
    }
}
